src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -1055,15 +1058,21 @@ type AdaptiveFSharpLspServer
             >> Log.addContextDestructured "file" file
           )
 
-          use s = File.openFileStreamForReadingAsync file
+          if File.Exists(UMX.untag file) then
+            use s = File.openFileStreamForReadingAsync file
 
-          let! source = sourceTextFactory.Create(file, s) |> Async.AwaitCancellableValueTask
+            let! source = sourceTextFactory.Create(file, s) |> Async.AwaitCancellableValueTask
 
-          return
-            { LastTouched = File.getLastWriteTimeOrDefaultNow file
-              Source = source
-              Version = 0 }
+            return
+              { LastTouched = File.getLastWriteTimeOrDefaultNow file
+                Source = source
+                Version = 0 }
 
+          else // When a user does "File -> New Text File -> Select a language -> F#" without saving, the file won't exist
+            return
```

Comment:
Hmm this file should be added to the `openFiles` and shouldn't be getting here.  Probably better to return `None` instead

Reply:
There will still be warnings written to VSCode output if I return None here. Returning this value, meanwhile, doesn't result in warnings written to VSCode output.